### PR TITLE
parser, ddl: update the DDL job's Location structure

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31212

Problem Summary:
The Location in the job is empty after decoding.

### What is changed and how it works?
Because the `Location` object's fields are all lowercase, there are no useful fields after json decoding. So, use `Name` and `Offset` instead of `Location` to represent timezone.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]
LGTM

File: parser/model/ddl.go [hunk 1]
```diff
@@ -211,7 +211,13 @@ type DDLReorgMeta struct {
 	SQLMode       mysql.SQLMode                    `json:"sql_mode"`
 	Warnings      map[errors.ErrorID]*terror.Error `json:"warnings"`
 	WarningsCount map[errors.ErrorID]int64         `json:"warnings_count"`
-	Location      *time.Location                   `json:"time_location"`
+	Location      *TimeZone                        `json:"time_zone"`
```
Comment (wjhuang2016) [inline parser/model/ddl.go:214]: Would it break the compatibility?

Comment (zimulala) [reply parser/model/ddl.go:214]: No, I looked at the original PR that added this field that is not on the branch yet.

Comment (wjhuang2016) [reply parser/model/ddl.go:214]: Which PR?

Comment (zimulala) [reply parser/model/ddl.go:214]: PR：https://github.com/pingcap/tidb/pull/29664

**Review by wjhuang2016 — COMMENTED** [review]
Need an issue

File: parser/model/ddl.go (hunk 1 above)
Comment (wjhuang2016) [inline parser/model/ddl.go:214]: Where do you set it at?

Comment (zimulala) [reply parser/model/ddl.go:214]: I will set it in the next PR（There are a lot of problems involved in this PR, so I will fix this problem first
）.

Comment (wjhuang2016) [reply parser/model/ddl.go:214]: But I don't know what you want to fix and don't know if it works.

Comment (zimulala) [reply parser/model/ddl.go:214]: It describes the problems we need to deal with: “The `Location` in the job is empty after decoding”.  It was also written in the original PR of worker to be dealt with.

Comment (wjhuang2016) [reply parser/model/ddl.go:214]: I mean the problem from the user's perspective. 
“The Location in the job is empty after decoding”, Sounds like a problem, but what would happen? I'm still confused.
Besides, why it's empty after decoding? And Why using `Name` and `Offset` can fix it?

Comment (zimulala) [reply parser/model/ddl.go:214]: Because the Location object's fields are all lowercase, there are no useful fields after decoding.

Comment (wjhuang2016) [reply parser/model/ddl.go:214]: Ok, I get it. Please add an issue for this PR.

Comment (zimulala) [reply parser/model/ddl.go:214]: @wjhuang2016 Done.

**Comment by zimulala** [comment]
CI failed for：
```
[2021-12-31T03:36:23.943Z] FAIL: update_test.go:1446: testStatsSuite.TestLogDetailedInfo
[2021-12-31T03:36:23.943Z] 
[2021-12-31T03:36:23.943Z] update_test.go:1507:
[2021-12-31T03:36:23.943Z]     c.Assert(s.hook.results, Equals, t.result)
[2021-12-31T03:36:23.943Z] ... obtained string = ""
[2021-12-31T03:36:23.943Z] ... expected string = "[stats-feedback] test.t, column=a, rangeStr=range: [-inf,8), actual: 8, expected: 8, buckets: {num: 8 lower_bound: 0 upper_bound: 7 repeats: 1 ndv: 0, num: 8 lower_bound: 8 upper_bound: 15 repeats: 1 ndv: 0}[stats-feedback] test.t, column=a, rangeStr=range: [8,15), actual: 8, expected: 7, buckets: {num: 8 lower_bound: 8 upper_bound: 15 repeats: 1 ndv: 0}"
```

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**